### PR TITLE
Replacing deprecated constants, adding and fixing stories

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR is a part of the following issues
  - #19526 
  - #18896 
  - #18714 
  - #19555
 
Replaced deprecated consts in the following folders

- `ui/components/app/snaps/update-snap-permission-list`
- `ui/components/app/snaps/snap-privacy-warning`
- `ui/components/app/snaps/snap-settings-card`
- Created Story for `SnapInstallWarning`
- Fixed story error for `SnapSettingsCard`

### Before
image

### After
image
image

#### `targetedMetadata` propType [error](https://github.com/MetaMask/metamask-extension/pull/19584#discussion_r1230180132) fixed
image

### Creating Story for `SnapInstallWarning`
image
image

Fixing the [error](https://github.com/MetaMask/metamask-extension/pull/19584#issuecomment-1592017104) while trying to click outside the Box

https://github.com/MetaMask/metamask-extension/assets/79097544/8d420824-1fbe-4035-aefe-667c07bafe08

### Before `UpdateSnapPermissionList`
image
image
### After `UpdateSnapPermissionList`
image
image

### Before `SnapPrivacyWarning`
image
image
### After `SnapPrivacyWarning`
image
image

### Before `SnapUpdate`
image
image
image
image
image

### After `SnapUpdate`
image
image
image
image
image

### Before `SnapConnect`
image
image
image
image

### After `SnapConnect`
image
image
image

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, apologies for the number of files included in the PR. Initially, I had only made changes to 4 files but when I was going through the page for Before screenshots I found an error in `SnapSettingCard` story so I tried fixing it while I was at it. I hope you dont mind. I'll make sure to open separate PRs next time. Apologies for the inconvenience.
Thanks!

File: ui/components/app/snaps/snap-install-warning/index.scss
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-install-warning/index.scss:1]: Removing unneeded CSS

**Review by georgewrmarshall — COMMENTED** [review]
Hey @dhruvv173, great updates! I saw some more deprecated components that could be replaced

https://github.com/MetaMask/metamask-extension/issues/19555

Would you checking my updates on your end and updating the screenshots in you PR?

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, great updates! I saw some more deprecated components that could be replaced
> 
> https://github.com/MetaMask/metamask-extension/issues/19555
> 
> Would you checking my updates on your end and updating the screenshots in you PR?
> 

No worries, I'll replace the remaining deprecated components and update the screenshots.

File: ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.stories.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import UpdateSnapPermissionList from './update-snap-permission-list';
 
 export default {
-  title: 'Components/App/UpdateSnapPermissionList',
+  title: 'Components/App/Snaps/UpdateSnapPermissionList',
   component: UpdateSnapPermissionList,
```
Comment (dhruvv173) [inline ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.stories.js:7]: as the `targetedSubjectMetadata` is not passed in the story it throws an error in console
can the object be passed as the one as pasted below?
```
{
    name: PropTypes.string,
    origin: PropTypes.string.isRequired,
    subjectType: PropTypes.string.isRequired,
    extensionId: PropTypes.string,
    iconUrl: PropTypes.string,
}
```

**Comment by dhruvv173** [comment]
hey @georgewrmarshall I have updated the screenshots in the PR description and fixed the small lint issue.
Found one error in `SnapInstallWarning` story while trying to click outside the warning box. Is this behaviour expected or its an issue? would appreciate your insight.
image
image

File: ui/components/app/snaps/snap-install-warning/snap-install-warning.stories.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export default {
+  title: 'Components/App/snaps/SnapInstallWarning',
+  component: SnapInstallWarning,
+  argTypes: {
+    onCancel: {
+      action: 'onCancel',
+    },
+    onSubmit: {
+      action: 'onSubmit',
+    },
```
Comment (georgewrmarshall) [inline ui/components/app/snaps/snap-install-warning/snap-install-warning.stories.js:13]: Function type props can use the action control. Updated here

Comment (dhruvv173) [reply ui/components/app/snaps/snap-install-warning/snap-install-warning.stories.js:13]: okay, I'll keep this in mind while adding new stories 👍

**Comment by dhruvv173** [comment]
hey @georgewrmarshall I have updated the PR description following your recent commits, please go through it once when possible. Let me know if anything else needs to be addressed otherwise this is ready for review imo.

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, it looks like the e2e tests are failing on this PR. Are you familiar with debugging e2e tests? It might be a good opportunity to become familiar with them. I think some of my updates removed an id selector. Would you be able add the id selector to the appropriate component so tests pass? You can read about e2e testing locally in the README

**Comment by dhruvv173** [comment]
hey @georgewrmarshall I encountered some issues while running the e2e tests locally, preventing me from identifying the root cause of the failing snaps tests. However, I tried to work around it by opening a draft PR and committing to it so that e2e tests can run on changed code. I understand that this is not the best practice and I apologize for using this approach :/
I think the failing snaps test might be related to `snap-install-warning` component, I might be wrong tho, would appreciate your insight on it.
Apologies for not being able to fix the issue :(

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, no worries I can walk you through it

Run
```
build:test:flask
```

Then run the test that is failing in Firefox

```
yarn test:e2e:single test/e2e/snaps/test-snap-bip-32.spec.js --browser=firefox --debug --leave-running
```

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, no worries I can walk you through it
> 
> Run
> 
> ```
> yarn build:test:flask
> ```
> 
> Then run the test that is failing in Firefox
> 
> ```
> yarn test:e2e:single test/e2e/snaps/test-snap-bip-32.spec.js --browser=firefox --debug --leave-running
> ```

thanks for the help! However, when I try to use the built extension I get this error in console and the extension doesnt start.  
```
Error: LavaMoat - required package not in allowlist: package "@metamask/eth-trezor-keyring>@trezor/connect-web>@trezor/connect" requested "browserify>events" as "events"
```
image

I found an open issue related to it [here](https://github.com/MetaMask/metamask-extension/issues/13253).
There was another similar issue which was related to building without lavamoat. The dev mode works fine for me but theres an issue with test build.

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, would it be possible to rebase this PR and get it up to date?

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, would it be possible to rebase this PR and get it up to date?

done!

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, I think these updates are great but to ensure we get them it would it be possible to split this PR up into smaller ones that way it will be easier for the snaps team to review. Thanks!

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, I think these updates are great but to ensure we get them it would it be possible to split this PR up into smaller ones that way it will be easier for the snaps team to review. Thanks!

okay will do that over the weekend, thank you for the update!

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, I think these updates are great but to ensure we get them it would it be possible to split this PR up into smaller ones that way it will be easier for the snaps team to review. Thanks!

Done in the below PRs:
- #20301 
- #20296 
- #20294 
- #20284

**Pull request closed without merging**